src/conv/invokers/mlir_impl_gemm.cpp

```diff
@@ -0,0 +1,153 @@
+#include <miopen/conv/invokers/mlir_impl_gemm.hpp>
+#include <miopen/memref.hpp>
+
+#include <miopen/conv/data_invoke_params.hpp>
+#include <miopen/algorithm.hpp>
+#include <miopen/handle.hpp>
+#include <miopen/tensor_ops.hpp>
+
+#include <boost/any.hpp>
+
+namespace miopen {
+namespace conv {
+
+namespace {
+using MemRef4DGeneric = StridedMemRefType<void, 4>;
+
+struct MlirConvArgs
+{
+    MemRef4DGeneric filter;
+    MemRef4DGeneric input;
+    MemRef4DGeneric output;
+};
+
+// Rearrange strides correctly
+// In MLIR: the layout, sizes and strides are coherent. The layout information is not
+// embedded into the permutation of strides.
+// - For NCHW, sizes = {N, C, H, W}; strides = {C*H*W, H*W, W, 1}
+// - For NHWC, sizes = {N, H, W, C}; strides = {C*H*W, W*C, C, 1}
```

Comment:
If MLIR doesn't use the strides to describe the layout of the tensor, then should we even pass the strides at all to MLIR?

Reply:
I have experimented to test with strides of all zeros and confirm that it can run well, so yes, that confirms strides are not used for now.

I do this mostly for robustness:
 - I can't guarantee future development of MLIR will not use this field
 - All MLIR local testing have the complete memref testing. In other words, incomplete type is not track in CI.